scapy/layers/dns.py

```diff
@@ -462,8 +462,8 @@ def RRlist2bitmap(lst):
         for tmp in xrange(bytes):
             v = 0
             # Remove out of range Resource Records
-            tmp_rrlist = filter(lambda x: 256 * wb + 8 * tmp <= x < 256 * wb + 8 * tmp + 8, rrlist)
-            if tmp_rrlist:
+            tmp_rrlist = [x for x in rrlist if 256 * wb + 8 * tmp <= x < 256 * wb + 8 * tmp + 8]
+            if not tmp_rrlist == []:
```

Comment:
The test `if not tmp_rrlist == []:` should be `if tmp_rrlist:`.
Also, I'm sorry but you're still building unnecessary lists.
I really think the most Pythonic and efficient way to write this bloc is to use a `"".join()` construction.